### PR TITLE
[Dependency] PyPDF2 version update

### PR DESCRIPTION
**Pull request by akshatbatra**
### What is the current behavior?
Related to #7289, some PDF files are not accepted by PyPDF2.

### What is the new behavior?
Active maintenance of PyPDF2 has resulted in it being more dependable while working with corrupt PDFs.
PyPDF2 is now able to work with files such as [broken.pdf](https://github.com/Submitty/Submitty/files/7422603/broken.pdf) with `strict=False` set, it can now deal with xref table not being present at the specified location.
Required version of PyPDF2 has been updated to 1.28.2 and its deprecated usage has been replaced with latest recommendations.

### CODE REVIEW

**Comment by cjreed121** [comment]
This is a duplicate of https://github.com/Submitty/Submitty/pull/7926. One of these should be closed.

**Comment by bmcutler** [comment]
Yes please, make a PR to the other repo :)

**Review by Eceptonsu — APPROVED** [review]
It works well for me even though it is a bit slow.

**Review by shailpatels — APPROVED** [review]
Thanks for the fix and good to hear that pypdf2 is under active maintenance again!

Tested with regular and bulk upload and both appeared to work as expected

File: sbin/submitty_daemon_jobs/submitty_jobs/bulk_qr_split.py [hunk 1]
```diff
@@ -33,15 +33,15 @@ def main(args):
 
     try:
         os.chdir(split_path)
-        pdfPages = PdfFileReader(filename)
+        pdfPages = PdfFileReader(filename, strict=False)
         pdf_writer = PdfFileWriter()
```
Comment (MasterOdin) [inline sbin/submitty_daemon_jobs/submitty_jobs/bulk_qr_split.py:37]: We'll need to update the class names while we're at it as the `PdfFile*` version has been deprecated in favor of the shorter names.

```suggestion
        pdfPages = PdfReader(filename, strict=False)
        pdf_writer = PdfWriter()
```

**Review by MasterOdin — CHANGES_REQUESTED** [review]

**Review by MasterOdin — APPROVED** [review]

**Comment by MasterOdin** [comment]
Before merging this, we need to improve how we manage python dependencies so that they get updated as part of running `INSTALL_SUBMITTY_HELPER.sh` instead of on `install_system.sh`, as otherwise all code that relies on the new version of PyPDF2 will break until manually run `pip install -U PyPDF2` (with appropriate chmod).

**Pull request merged**